### DISCUSSION TITLE
Realtime - Connections

### DISCUSSION THREAD

**Discussion by go4cas** [pocketbase/pocketbase#5101] [Q&A]
My use case: maintaining a list of active realtime connections. I am using JavaScript to extend my PB instance.

Then, I have a few questions (apologies if I have missed these in the docs):

1. Reading through some of the discussions, it seems that the best option for my use case is to create a collection, and maintain a record for each connection. Is this still the best approach?
2. Which event hook would be the best to add new connections to the collection? `onRealtimeConnectRequest` or `onRealtimeBeforeSubscribeRequest` or `onRealtimeAfterSubscribeRequest `?
3. Ideally I would like to capture some meta data with the connection id - in my case, specifically the client IP Address and the User-Agent header. How can I get these from within the event hook? I have tried `e.httpContext`, but this seems to always be empty.

Thanks again for the awesome project!

**Comment by ganigeorgiev** [comment]
The description is too vague and I don't know what to recommend as it is not clear what exactly do you mean by _"maintaining a list of active realtime connections"_.

If you trying to create "presence" like functionality, then it would depend based on what criteria you want to count a user as "present", how your application will use it and who must have access to that information (for example a "present" could mean that user that has established any realtime connection to the PocketBase server or only if the user has subscribed to a specific topic/room/channel).

There are probably dozens other similar questions posted here and to the support mail (https://github.com/pocketbase/pocketbase/discussions/1512, https://github.com/pocketbase/pocketbase/discussions/3437, https://github.com/pocketbase/pocketbase/discussions/3378, https://github.com/pocketbase/pocketbase/discussions/845, https://github.com/pocketbase/pocketbase/discussions/1607, https://github.com/pocketbase/pocketbase/discussions/1706 and probably others) but the common thing between them is that there is no single solution for this problem because it depends on your actual use case and users fail/refuse to elaborate on this and I don't know how to help or make it more simple (_see also [XY problem](https://xyproblem.info/)_).

> I have tried e.httpContext, but this seems to always be empty.

If you are just printing `console.log(e.httpContext)` then you'll not see the available fields because `e.httpContext` is an interface (_I'll update the documentation in the future to remove such examples to avoid the confusion_), but this doesn't mean that it is "empty". Most of the JSVM fields are not native JS objects (they are direct translations from Go) and may not have public serializable fields or safe string representation to show. Please refer to https://pocketbase.io/jsvm/interfaces/echo.Context.html and the general [JS Routing docs](https://pocketbase.io/docs/js-routing) (`c` in the examples is the same as `e.httpContext`).
